packages/babel-generator/src/node/parentheses.js

```diff
@@ -55,6 +55,13 @@ export function ObjectExpression(node: Object, parent: Object): boolean {
     return true;
   }
 
+  if ((t.isBinaryExpression(parent) || t.isLogicalExpression(parent)) && parent.left === node) {
+    // We'd need to check that the parent's parent is an ExpressionStatement. But this
```

Comment:
Yeah, it's legal in those cases. The only time that it's illegal is that when it's `Binary|LogicalExpression` inside an `ExpressionStatement`:

bad: `{} === 1`